### PR TITLE
atari/atarist.cpp: enable floppy sounds

### PR DESCRIPTION
**Pull request by angelosa**

### CODE REVIEW

File: src/mame/atari/atarist.cpp [hunk 1]
```diff
@@ -1605,8 +1605,8 @@ void stbook_state::stbook(machine_config &config)
 	WD1772(config, m_fdc, U517/2);
 	m_fdc->intrq_wr_callback().set(m_mfp, FUNC(mc68901_device::i5_w)).invert();
 	m_fdc->drq_wr_callback().set(FUNC(st_state::fdc_drq_w));
-	FLOPPY_CONNECTOR(config, WD1772_TAG ":0", atari_floppies, "35dd", 0, st_state::floppy_formats);
-	FLOPPY_CONNECTOR(config, WD1772_TAG ":1", atari_floppies, 0,      0, st_state::floppy_formats);
+	FLOPPY_CONNECTOR(config, WD1772_TAG ":0", atari_floppies, "35dd", 0, st_state::floppy_formats).enable_sound(true);
+	FLOPPY_CONNECTOR(config, WD1772_TAG ":1", atari_floppies, 0,      0, st_state::floppy_formats).enable_sound(true);
```
Comment (angelosa) [inline src/mame/atari/atarist.cpp:1609]: I believe the 0 instead of `nullptr` for slot type won't compile, but this block is disabled under `#if 0`.

**Comment by galibert** [comment]
I hate the current implementation of floppy sounds, so don't ask me.

**Comment by angelosa** [comment]
I see it's not perfect, but having audio cues is useful for debugging, plus you can always disable them in sliders menu.
Also, I'm trying not to stomp your feet, i.e. if you have WIP work in this file that may conflict.

**Comment by galibert** [comment]
No, I don't think there's a risk on conflict on that patch at this point in time.

**Comment by cuavas** [comment]
I really dislike the floppy sounds – they aren’t even close to realistic, and they mess up recording sound output, e.g. with `-aviwrite`, particularly for demos that do background loading.  It’s like a spreading cancer.

**Comment by angelosa** [comment]
Well, again: let's make a `mame.ini` option that opts out of the floppy sounds? It's useful for debugging, otherwise there's no floppy (video/)audio cue unless you resort to recompiling.

**Comment by Gurudumps** [comment]
> Well, again: let's make a `mame.ini` option that opts out of the floppy sounds? It's useful for normal debugging, otherwise there's no floppy (video/)audio cue unless you resort to recompiling.

Yes. I made that global mame ini suggestion at least 3-4 years ago. What do people have against doing it the right way? Looks like no one can decide so as usual nothing is done :-/

**Comment by cuavas** [comment]
If the use case is user-visible feedback for disk drive activity, the proper solution is scoping output names by device so the floppy drive devices can have outputs for motor state, head position, etc. that don’t stomp on each other as soon as you have multiple drives in the system.  That requires people to keep replacing uses of `output_manager::set_value` with output finders.

**Pull request closed without merging**